client/components/translator-invite/index.jsx

```diff
@@ -0,0 +1,132 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import { startsWith } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import QueryLanguageNames from 'components/data/query-language-names';
+import getLocalizedLanguageNames from 'state/selectors/get-localized-language-names';
+import { getLanguage, getLocaleFromPath } from 'lib/i18n-utils';
+import { recordTracksEvent } from 'state/analytics/actions';
+/**
+ * Module variables
+ */
+const defaultLanguageSlug = config( 'i18n_default_locale_slug' );
+
+export class TranslatorInvite extends Component {
+	static propTypes = {
+		locale: PropTypes.string,
+		path: PropTypes.string.isRequired,
+	};
+
+	static defaultProps = {
+		locale: '',
+		path: '',
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			locale: this.getLocale(),
```

Comment:
I personally found that having `props.locale` and `state.locale` at the same time a bit confusing. From the code logic, `state.locale` should be dependent on `props.locale`, but updating `props.locale` doesn't really update `state.locale` after component constructing stage. It's also not something we would want to update via `setState()`. With these two reasons, I personally don't think it fits into the component state.

I'm not exactly sure how at the moment, but I'd propose us to find a way that 

1. Doesn't rely on the component state, since it's not a mutating state.
1. Updating `props.locale` should also affect this property.

Maybe we could move the logic of `getLocale()` function to the react-redux `connect()` call below and make `props.locale` replace the current `state.locale`.

Reply:
Agree, and thanks for pointing it out. It's not really a selector, but it's worth abstracting. Maybe a `util` function that accepts `props.locale` and `props.path`.